src/shortestpaths/floyd-warshall.jl

```diff
@@ -57,18 +79,31 @@ function floyd_warshall_shortest_paths(
             d == typemax(T) && continue
             p = parents[pivot, v]
             for u in vertices(g)
-                ans = (dists[u, pivot] == typemax(T) ? typemax(T) : dists[u, pivot] + d) 
+                ans = (dists[u, pivot] == typemax(T) ? typemax(T) : dists[u, pivot] + d)
                 if dists[u, v] > ans
                     dists[u, v] = ans
                     parents[u, v] = p
                 end
             end
         end
     end
+
+    #If dists[i,j] is negative, it means that there is a negative cycle in going from i to j
+    @inbounds for i=1:nvg
+        @inbounds for j=1:nvg
+            @inbounds for t = 1:nvg
```

Comment:
you don't need `@inbounds` for all three loops, and it might be better to make a single loop here. Also, you might be wanting to use `vertices(g)` instead.

Reply:
Waitasec. This also makes this algorithm O(|V|^3) if I'm reading this right. Is that the case?

Reply:
Yes, this an O(|V|^3) algorithm. Floyd Warshall in itself is also O(|V|^3), so by adding these lines the asymptotic complexity doesn't change. 
This is the only way to make this correction because we have to check for all other vertices `t` for all combinations of `i` and `j`.

Reply:
I'd like to see some before-and-after benchmarks on large (million-vertex) graphs. Could you do that?

Reply:
Yeah, I can do the benchmarking but normal Floyd Warshall will itself take (10^6)^3 = (10^18) iterations which will take neary 10^10 sec or 16666667 hrs to run so that is not feasible.
I will do the benchmarking for `nvg=10^3` because that will take only 10^9 operations and would run in about 10 sec.

I'm making one improvement, please see my next commit for that. That will help in running my O(|V|^3) algorithm only when there is a negative weight cycle in graph.

Reply:
Of course. Sorry for thr brain freeze on the magnitude of |V|. But see below. I don’t know that we want to legitimize negative edge weights.

Reply:
Graph with negative weights are quite common.
I found these while searching for real life applications of negative weight edges :      
https://stackoverflow.com/questions/8624468/negative-edge-in-graph
https://stackoverflow.com/questions/7576519/negative-weight-graph-edges

Reply:
Other graph libraries also allow working with negative weight edges, I am not saying we should copy them but graphs with negative weight edges and cycles look quite ok to me.